server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/CommitWatchers.java

```diff
@@ -123,7 +129,12 @@ void close(Supplier<CentralDogmaException> causeSupplier) {
             for (final Set<Watch> watches : watchesMap.values()) {
                 for (final Iterator<Watch> i = watches.iterator(); i.hasNext();) {
                     final Watch w = i.next();
-                    eligibleWatches = move(eligibleWatches, i, w);
+                    if (w.future() == null) {
```

Comment:
```suggestion
                    if (!w.canRemove()) {
```

Reply:
I thought `future() == null` was more expressive in conveying the current state, but using  `canRemove()` also sounds good.